analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -309,4 +313,30 @@ private static ProgramState InitLocals(ControlFlowBranch branch, ProgramState st
         region.Kind == ControlFlowRegionKind.Finally
         || thrown == ExceptionState.UnknownException
         || thrown.Type.DerivesFrom(region.ExceptionType);
+
+    private static IEnumerable<int> DetectLoopBlockOrdinals(ControlFlowGraph cfg)
```

Comment:
Educational: Can you give me the elevator pitch on why we need this?

Reply:
The goal is to process operations differently based on whether or not they are part of a loop. If they are in a loop, we must be more careful with our learnings since we only execute two loop iterations.